agent/engine/dockerclient/dockerclientfactory.go

```diff
@@ -14,13 +14,21 @@
 package dockerclient
 
 import (
-	"errors"
-
 	"github.com/aws/amazon-ecs-agent/agent/engine/dockeriface"
+	"github.com/aws/amazon-ecs-agent/agent/utils"
 	log "github.com/cihub/seelog"
 	docker "github.com/fsouza/go-dockerclient"
+	"github.com/pkg/errors"
 )
 
+const (
+	// minDockerAPILinux is the min Docker API version supported by agent
+	minDockerAPILinux = Version_1_17
```

Comment:
You need to specify this differently for Linux and Windows.  Windows won't respond at this API version.  You can change the constants in `versionsupport_windows.go` and `versionsupport_unix.go` to have the same name and then just use those.